test/e2e/framework/test_context.go

```diff
@@ -59,7 +59,7 @@ func ValidateFlags(t *TestContextType) {
 		klog.Fatalf("kubeconfig parameter or KUBECONFIG environment variable is required")
 	}
 
-	if len(t.KubeContexts) < 2 && len(t.KubeConfigs) < 2 {
-		klog.Fatalf("several kubernetes contexts are necessary east, west, etc..")
+	if len(t.KubeContexts) < 1 && len(t.KubeConfigs) < 1 {
```

Comment:
This is a problem in the other uses where we always expect clusterA & clusterB

On other E2E framework usages the user will be able to pass only one context, and it will panic when trying to access the index for clusterB

Reply:
I think each user of this API needs to validate the context based on their use case. This code can't assume the number of clusters/contexts and certainly the prior check didn't work with only one cluster. So here we just make a basic check that at least 1 context must be specified.

Reply:
So shipyard e2e tests only need 1 cluster; admiral 2.

Reply:
Yeah, that would make sense.